pkg/util/kube/pvc_pv.go

```diff
@@ -41,9 +41,25 @@ const (
 	waitInternal = 2 * time.Second
 )
 
-// DeletePVCIfAny deletes a PVC by name if it exists, and log an error when the deletion fails
-func DeletePVCIfAny(ctx context.Context, pvcGetter corev1client.CoreV1Interface, pvcName string, pvcNamespace string, log logrus.FieldLogger) {
-	err := pvcGetter.PersistentVolumeClaims(pvcNamespace).Delete(ctx, pvcName, metav1.DeleteOptions{})
+// DeletePVAndPVCIfAny deletes PVC and delete the bound PV if it exists and log an error when the deletion fails
+// We first set the reclaim policy of the PV to Delete, then PV will be deleted automatically when PVC is deleted.
+func DeletePVAndPVCIfAny(ctx context.Context, client corev1client.CoreV1Interface, pvcName, pvcNamespace, pvName string, log logrus.FieldLogger) {
+	pv, err := client.PersistentVolumes().Get(ctx, pvName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) { // pv will be nil
+			log.WithError(err).Debugf("Abort deleting PV, it doesn't exist, %s", pvName)
+		} else {
+			log.WithError(err).Errorf("Failed to get PV %s", pvName)
+		}
+	}
+	if pv != nil {
+		_, err = SetPVReclaimPolicy(ctx, client, pv, corev1api.PersistentVolumeReclaimDelete)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to set reclaim policy of PV %s to Delete", pvName)
```

Comment:
Warning or Error? If we set it as Error log, the backup will be set to PartiallyFailed

Reply:
I will change it into Warning